### PR TITLE
Lawrence/rankings modifications

### PR DESCRIPTION
**Pull request by lawrencechen14**
- Fixed last_responded bug to now update when submitted preferences
- Now only shows back button in rankings edit page if logged in
- UI enhancements to rankings show and edit pages, as well as fixing merge conflict UI bug at the end page

### CODE REVIEW

File: app/controllers/rankings_controller.rb [hunk 1]
```diff
@@ -22,6 +24,28 @@ def valid_secret_id?
     params[:secret_id] == @participant.secret_id
   end
 
+  def create_rankings_hash
+    @rankings_hash = Hash.new
+    @rankings.each do |ranking|
+      unless ranking.project_time.is_date
```
Comment (kev-chien) [inline app/controllers/rankings_controller.rb:30]: nice! thanks for adding this check. also I like how you use `ranking.project_time`, good use of Rails convention!

File: app/controllers/rankings_controller.rb [hunk 2]
```diff
@@ -22,6 +24,28 @@ def valid_secret_id?
     params[:secret_id] == @participant.secret_id
   end
 
+  def create_rankings_hash
```
Comment (kev-chien) [inline app/controllers/rankings_controller.rb:27]: Using a hash seems like a good constant time solution for this. Shows that software engineering interviews do have some sort of point in testing big O!

File: app/controllers/rankings_controller.rb [hunk 3]
```diff
@@ -57,15 +69,18 @@ def create
...
+        existing_ranking.update(
+            :rank => rank_num, :participant_id => @participant.id, :project_time_id => id)
+      else
+        new_ranking = Ranking.new(:rank => rank_num, :participant_id => @participant.id, :project_time_id => id)
+        new_ranking.save!
       end
-      new_rank.save!
     end
 
+    @participant.update(last_responded: Time.now.getutc)
```
Comment (kev-chien) [inline app/controllers/rankings_controller.rb:87]: Thanks for fixing!

**Review by alined908 — COMMENTED** [review]
- Getting Error: please fill in an option for each time, when filling out preferences and submitting
- Would be good to have endtime/duration
- ui looks much better

**Comment by rejoycesong** [comment]
- UI changes look great
- Thanks for changing the back button!

**Review by kev-chien — COMMENTED** [review]
Good changes. Seems like the rankings don't fully work though (as said by other reviewers), can you look into it? Thanks

**Comment by kev-chien** [comment]
Still getting an "Error: please fill in an option for each time." when I try to submit preferences with a time given for each timeslot.

**Comment by lawrencechen14** [comment]
Do not approve yet, there's still some local changes needed to push

File: app/views/projects/show.html.erb [hunk 4]
```diff
@@ -6,9 +6,9 @@
 
   <div class="current-duration">
     <% if !@project.duration.blank? %>
-      <p><strong>Current Timeslot Duration(miniutes):</strong> <%= @project.duration %> minutes</p>
+      <p><strong>Current Timeslot Duration:</strong> <%= @project.duration %> minutes</p>
```
Comment (kev-chien) [inline app/views/projects/show.html.erb:9]: Thanks for fixing

**Review by kev-chien — APPROVED** [review]
Looks good to me! Behavior works as expected now. Looks ready to merge.

**Review by rejoycesong — APPROVED** [review]
Looks great, thanks for the fix.

**Pull request merged**